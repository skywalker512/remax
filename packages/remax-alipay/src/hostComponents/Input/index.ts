import * as React from 'react';
import createHostComponent from '../../createHostComponent';

export interface InputProps {
  readonly dataset?: DOMStringMap;
  id?: string;
  className?: string;
  value?: string;
  name?: string;
  type?: string;
  password?: boolean;
  placeholder?: string;
  placeholderStyle?: React.CSSProperties;
  placeholderClass?: string;
  disabled?: boolean;
  maxlength?: number;
  focus?: boolean;
  confirmType?: 'done' | 'go' | 'next' | 'search' | 'send';
  confirmHold?: boolean;
  cursor?: number;
  selectionStart?: number;
  selectionEnd?: number;
  randomNumber?: boolean;
  controlled?: boolean;
  onInput?: (e: any) => void;
  onConfirm?: (e: any) => void;
  onFocus?: (e: any) => void;
  onBlur?: (e: any) => void;
}
const Input = createHostComponent<InputProps>('input');

export default Input;
