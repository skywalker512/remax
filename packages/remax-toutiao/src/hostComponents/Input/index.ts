import * as React from 'react';

import { BaseProps } from '../../types/component';
import createHostComponent from '../../createHostComponent';

export interface InputProps extends BaseProps {
  focus?: boolean;
  maxlength?: number;
  value: string;
  password?: boolean;
  type: 'text' | 'number' | 'digit';
  cursor?: number;
  selectionStart?: number;
  selectionEnd?: number;
  adjustPosition?: boolean;
  placeholder?: string;
  placeholderStyle?: React.CSSProperties;
  placeholderClass?: string;
  disabled?: boolean;
  cursorSpacing?: number;
  onInput?: (e: any) => void | string;
  onClick?: (e: any) => void;
  onFocus?: (e: any) => void;
  onBlur?: (e: any) => void;
  onConfirm?: () => void;
}

export default createHostComponent<InputProps>('input');
