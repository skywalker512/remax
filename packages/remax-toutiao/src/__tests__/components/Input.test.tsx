import * as React from 'react';
import TestRenderer from 'react-test-renderer';
import Input from '../../hostComponents/Input';

describe('Input', () => {
  it('render correctly', () => {
    const tree = TestRenderer.create(<Input value="" type="text" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
