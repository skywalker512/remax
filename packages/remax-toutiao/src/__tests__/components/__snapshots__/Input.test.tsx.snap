// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render correctly 1`] = `
<input
  type="text"
  value=""
/>
`;
